flaml/autogen/agent/human_proxy_agent.py

```diff
@@ -0,0 +1,99 @@
+from .agent import Agent
+from flaml.autogen.code_utils import extract_code, execute_code
+
+
+class HumanProxyAgent(Agent):
+    """(Experimental) A proxy agent for human, that can execute code and provide feedback to the other agents."""
+
+    DEFAULT_SYSTEM_MESSAGE = """You are human agent. You can execute_code or give feedback to the sender.
+    """
+    MAX_TURN_NUM = 100  # maximum number of turns in one conversation session (subject to future change)
+
+    def __init__(
+        self,
+        name,
+        system_message="",
+        work_dir=None,
+        human_input_mode="ALWAYS",
+        max_turn_num=None,
+        is_termination_msg=None,
+        **config,
+    ):
+        """
+        Args:
+            name (str): name of the agent
+            system_message (str): system message to be sent to the agent
+            work_dir (str): working directory for the agent to execute code
+            human_input_mode (bool): whether to ask for human inputs every time a message is received.
+                Possible values are "ALWAYS", "TERMINATE", "NEVER".
+                When "ALWAYS", the agent will ask for human input every time a message is received.
+                When "TERMINATE", the agent will ask for human input only when a termination message is received.
+                When "NEVER", the agent will never ask for human input.
+            max_turn_num (int): the maximum number of turns in one conversation session.
```

Comment:
My understanding for this argument is the number of automatic turns.

Reply:
Revised. Thanks for the comment.